[ONNX] Enable 'ExportOutput.save' for models larger than 2GB

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107904

Previously it fails during serialization, despite onnxscript graph_building managed to return ModelProto > 2GB.

BowenBao committed:
[ONNX] Enable 'ExportOutput.save' for models larger than 2GB

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Enable 'ExportOutput.save' for models larger than 2GB"


Previously it fails during serialization, despite onnxscript graph_building managed to return ModelProto > 2GB.


[ghstack-poisoned]

BowenBao committed:
fix lint on "[ONNX] Enable 'ExportOutput.save' for models larger than 2GB"


Previously it fails during serialization, despite onnxscript graph_building managed to return ModelProto > 2GB.


[ghstack-poisoned]

abock commented on torch/onnx/_internal/exporter.py:
```diff
@@ -779,7 +812,12 @@ def save(
+                    serializer.serialize(self, destination)
+                except ValueError:
+                    raise ValueError(
+                        "'destination' must be a string when saving model larger than 2GB."
```
```python
"'destination' should be provided as a path-like string when saving a model larger than 2GB. External tensor data will be saved alongside the model on disk."
```

abock reviewed (approved):
This looks good for now. We should revisit the serializer protocol again for 2.2.

BowenBao committed:
Update on "[ONNX] Enable 'ExportOutput.save' for models larger than 2GB"


Previously it fails during serialization, despite onnxscript graph_building managed to return ModelProto > 2GB.


[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge
